common/containers/container.go

```diff
@@ -0,0 +1,296 @@
+		nextCont = nextCont[:i]
+		candidates = append(candidates, nextCont+"."+name)
+	}
+	candidates = append(candidates, name)
+	return c.candidatesWithAlias(candidates, name)
+}
+
+// aliasSet returns the alias to fully-qualified name mapping stored in the container.
+func (c *Container) aliasSet() map[string]string {
+	if c == nil || c.aliases == nil {
+		return noAliases
+	}
+	return c.aliases
+}
+
+// candidatesWithAlias returns the resolved candidates from the container with any applicable
+// alias appended on the end of the list.
+func (c *Container) candidatesWithAlias(candidates []string, name string) []string {
+	// If an alias exists for the name, ensure it is searched last.
+	alias, found := c.aliasSet()[name]
+	if found {
+		return append(candidates, alias)
+	}
+	return candidates
+}
+
+// ContainerOption specifies a functional configuration option for a Container.
+//
+// Note, ContainerOption implementations must be able to handle nil container inputs.
+type ContainerOption func(*Container) (*Container, error)
+
+// Abbrevs configures a set of simple names as abbreviations for fully-qualified names.
+//
+// An alias can be useful when working with variables, functions, and especially types from
+// multiple namespaces:
+//
+//    // CEL object construction
+//    qual.pkg.version.ObjTypeName{
+//       field: alt.container.ver.FieldTypeName{value: ...}
+//    }
+//
+// Only one qualified path may be used as CEL container, so at least one of these references is
+// a long qualified name within an otherwise short CEL program. Using the following abbreviations,
+// the program becomes much simpler:
+//
+//    // CEL Go option
+//    Abbrevs("qual.pkg.version.ObjTypeName", "alt.container.ver.FieldTypeName")
+//    // Simplified Object construction
+//    ObjTypeName{field: FieldTypeName{value: ...}}
+//
+// There are a few rules for the qualified names and the simple abbreviations generated from them:
+// - Qualified names must be dot-delimited, e.g. `package.subpkg.name`.
+// - The last element in the qualified name is the simple name used as the abbreviation.
+// - Abbreviations names must not collide with each other.
+// - The abbreviation name must not collide with the root-level container name.
+//
+// Abbrevs are distinct from container-based references in the following important ways:
+// - Containers follow C++ namespace resolution rules with searches from the most qualified name
+//   to the least qualified name.
+// - Container references within the CEL program may be relative, and are resolved to fully
+//   qualified names at either type-check time or program plan time, whichever comes first.
+// - Abbreviations must resolve to a fully-qualified name.
+// - Resolved abbreviations do not participate in namespace resolution.
+// - Resolved abbreviations are searched after container names, including container names in the
+//   global scope.
```

Comment:
I'm not sure what "container names in the global scope" means. Unless you want to introduce new terminology, "container name" sounds like it should be the name of the container. Maybe "containerized name"? Consider rewriting this bullet item as: "Abbreviation resolution happens after the container path has been searched for matching identifiers, including the global scope."

Reply:
Done.